test/mocked/raidenInspector.test.ts

```diff
@@ -0,0 +1,236 @@
+    });
+
+    it("throws for incorrect channel identifier", async () => {
+        const appointment = new RaidenAppointment({
+            ...appointmentObj,
+            stateUpdate: { ...appointmentObj.stateUpdate, channel_identifier: 2 }
+        });
+
+        const inspector = new RaidenInspector(minimumDisputePeriod, provider);
+        await isRejected(inspector.checkInspection(appointment));
+    });
+
+    it("throws for incorrect closing participant", async () => {
+        // correct closing participant: 0xdddEC4D561eE68F37855fa3245Cb878b10Eb1fA0
+        // use the other party: 0xccca21b97b27DefC210f01A7e64119A784424D26
+        const wrongClosingParticipant = "0xccca21b97b27DefC210f01A7e64119A784424D26";
+        const appointment = new RaidenAppointment({
+            ...appointmentObj,
+            stateUpdate: { ...appointmentObj.stateUpdate, closing_participant: wrongClosingParticipant }
+        });
+
+        const inspector = new RaidenInspector(minimumDisputePeriod, provider);
+        await isRejected(inspector.checkInspection(appointment));
+    });
+
+    it("throws for wrong closing participant sig", async () => {
+        // use sig by non closing party
+        const incorrectClosingParticipantSig =
+            "0xd77e9816a50c897a36dd02380b4789a07adb58f7a9cf91cf8c36447539a6bb6e1d118b035886e317189610dd3cdc8adbaeea4a78e5391876ba606189c1aeadd81b";
+        const appointment = new RaidenAppointment({
+            ...appointmentObj,
+            stateUpdate: { ...appointmentObj.stateUpdate, closing_signature: incorrectClosingParticipantSig }
+        });
+
+        const inspector = new RaidenInspector(minimumDisputePeriod, provider);
+        await isRejected(inspector.checkInspection(appointment));
+    });
+
+    it("throws for incorrect non_closing participant", async () => {
+        const wrongNonClosingParticipant = "0xdddEC4D561eE68F37855fa3245Cb878b10Eb1fA0";
+
+        const appointment = new RaidenAppointment({
+            ...appointmentObj,
+            stateUpdate: { ...appointmentObj.stateUpdate, non_closing_participant: wrongNonClosingParticipant }
+        });
+        const inspector = new RaidenInspector(minimumDisputePeriod, provider);
+        await isRejected(inspector.checkInspection(appointment));
+    });
+
+    it("throws for wrong non_closing participant sig", async () => {
+        // use sig by non closing party
+        const incorrectNonClosingParticipantSig =
+            "0x078d3465de0781b3011323df87e55491124fecdb74df8ed5857bdedf63f9a242680afc7311d0681b09108c01f3f72b9fcdffd803bced1df1fb944e219588286a1b";
+
+        const appointment = new RaidenAppointment({
+            ...appointmentObj,
+            stateUpdate: { ...appointmentObj.stateUpdate, non_closing_signature: incorrectNonClosingParticipantSig }
+        });
+
+        const inspector = new RaidenInspector(minimumDisputePeriod, provider);
+        await isRejected(inspector.checkInspection(appointment));
+    });
+
```

Comment:
Should we add a couple of tests that checksummed addresses are accepted in the appointment? (More general question: do we want to force users to provide checksummed addresses in our API?)

Reply:
We should add those tests, but currently that code exists in the appointment itself. We need to write a bunch of separate tests for the parsing code that exists there.

I would suggest that we probably should force the check sum, the more constrained the data is the better generally.